### PR TITLE
Improved blocks.insert() by allowing passing parameters as an object …

### LINKED ISSUE #1817: API: improve block.insert() interface to allow passing parameters as an object instead of set
1. Describe a problem.

The current interface of the `api.blocks.insert` accepts parameters as a list:

```ts
this.api.blocks.insert(
      toolName,
      undefined,
      undefined,
      index,
      undefined,
      currentBlock.isEmpty
);    
```

2. Describe the solution you'd like. Mockups are welcome.

It would better to pass parameters as an object:

```ts
this.api.blocks.insert({
  type: toolName,
  index,
  replace: currentBlock.isEmpty
});    
```

**Important**. We need to save the backward capability.

### PR DESCRIPTION
**Pull request by zang77**
* Improved block.insert() by allowing passing parameters as an object

* Fixed some issues commented out from last pr. Set the default parameter for insert.

* Added changelog for insert function.

Resolves #1817

### CODE REVIEW

**Comment by aditink** [comment]
Fixes issue #1817

File: src/components/modules/api/blocks.ts [hunk 1]
```diff
@@ -221,29 +221,23 @@ export default class BlocksAPI extends Module {
...
    *
-   * @param {string} type — Tool name
-   * @param {BlockToolData} data — Tool data to insert
-   * @param {ToolConfig} config — Tool config
-   * @param {number?} index — index where to insert new Block
-   * @param {boolean?} needToFocus - flag to focus inserted Block
-   * @param replace - pass true to replace the Block existed under passed index
+   * @param {NewBlock} newBlock — the block that will be inserted
    */
   public insert = (
```
Comment (neSpecc) [inline src/components/modules/api/blocks.ts:228]: it would be better to save the backward capability

File: types/api/block.d.ts [hunk 2]
```diff
@@ -74,3 +74,39 @@ export interface BlockAPI {
    */
   dispatchChange(): void;
 }
+
+/**
+ * @interface NewBlock Describes new block configuration
+ */
+
+export interface NewBlock {
```
Comment (neSpecc) [inline types/api/block.d.ts:82]:
```suggestion
export interface BlockInsertionParams {
```

**Review by neSpecc — CHANGES_REQUESTED** [review]

**Comment by neSpecc** [comment]
The solution should save the backward compatibility, but this PR is dropping it.

**Pull request closed without merging**